### ISSUE TITLE
Logs are not descriptive enough

### ISSUE THREAD

**Issue by darienalvarez** [cbeust/klaxon#220]
When you try to parse a JSON and some mandatory field is missing you get a null pointer exception but you never know what is the missing field or why the parsing fails

**Comment by cbeust** [comment]
Can you include an example of JSON and stack trace?

There is some rudimentary logging in Klaxon but it's not configurable yet (I will add this from the Klaxon object) and I can definitely improve the error reporting this way too.

**Comment by m4x-h4xx** [comment]
This is about the same troubles that we are currently facing, so I'll hijack it to provide a very related example.

To clarify, we were facing troubles when the data class that is to be de-serialized has a field with incompatible type. What we would really appreciate to see is more detail in the exception raised. From my understanding, it would not be too difficult to enrich the exception with information about

1. What field was missing, if such
2. Which field failed to get converted to the appropriate type

As you already mentioned, configurable logging would also be highly appreciated!

See below for an example for reproduction:

```kotlin
package somepackage

import com.beust.klaxon.Converter
import com.beust.klaxon.JsonValue
import com.beust.klaxon.Klaxon
import com.beust.klaxon.KlaxonException
import org.assertj.core.api.Assertions.assertThat
import org.assertj.core.api.Assertions.assertThatThrownBy
import org.junit.jupiter.api.Test
import java.time.LocalDate

val dateTime = object : Converter {
    override fun canConvert(cls: Class<*>): Boolean = cls == LocalDate::class.java
    override fun fromJson(jv: JsonValue): LocalDate = LocalDate.parse(jv.string)
    override fun toJson(value: Any): String = "\"$value\""
}

val klaxon = Klaxon().apply {
    converter(dateTime)
}

val referenceTestFixture = """
    {
      "lastName": "Doe",
      "firstName": "John",
      "dateOfBirth": "1990-11-23",
      "numberOfEyes": 2
    }
""".trimIndent()

val referenceFailingTestFixture = """
    {
      "lastName": "Doe",
      "firstName": "Jane",
      "dateOfBirth": "1990-11-23",
      "numberOfEyes": "2"
    }
""".trimIndent()

class ParseTypeFailureTest {

    @Test
    fun `it succeeds to parse a valid fixture`() {
        val johnDoe = klaxon.parse<Person>(referenceTestFixture)
        assertThat(johnDoe!!.numberOfEyes).isGreaterThan(0)
    }

    @Test
    fun `it fails to parse to the desired type when a field's type does not match`() {
        assertThatThrownBy {
            try {
                val janeDoe = klaxon.parse<Person>(referenceFailingTestFixture)
                assertThat(janeDoe!!.numberOfEyes).isGreaterThan(0)
            } catch (e: Exception) {
                e.printStackTrace()
                throw e
            }
        }.isOfAnyClassIn(KlaxonException::class.java)
    }
}

data class Person(
    val lastName: String,
    val firstName: String,
    val dateOfBirth: LocalDate,
    val numberOfEyes: Int
)
```

This would yield the following error & (partial) stacktrace:
```stacktrace
com.beust.klaxon.KlaxonException: Unable to instantiate Person with parameters [lastName: Doe, dateOfBirth: 1990-11-23, numberOfEyes: 2, firstName: Jane] - argument type mismatch
	at com.beust.klaxon.JsonObjectConverter.initIntoUserClass(JsonObjectConverter.kt:98)
	at com.beust.klaxon.JsonObjectConverter.fromJson(JsonObjectConverter.kt:32)
	at com.beust.klaxon.DefaultConverter.fromJsonObject(DefaultConverter.kt:214)
	at com.beust.klaxon.DefaultConverter.fromJson(DefaultConverter.kt:36)
	at com.beust.klaxon.Klaxon.fromJsonObject(Klaxon.kt:296)
	at klaxon.ParseTypeFailureTest$it fails to parse to the desired type when a field's type does not match$1.call(ParseTypeFailureTest.kt:73)
	at org.assertj.core.api.ThrowableAssert.catchThrowable(ThrowableAssert.java:62)
	at org.assertj.core.api.AssertionsForClassTypes.catchThrowable(AssertionsForClassTypes.java:739)
	at org.assertj.core.api.Assertions.catchThrowable(Assertions.java:1166)
	at org.assertj.core.api.Assertions.assertThatThrownBy(Assertions.java:1038)
	at klaxon.ParseTypeFailureTest.it fails to parse to the desired type when a field's type does not 
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:122)
	at java.base/java.lang.Thread.run(Thread.java:834)
```


That said, please keep up the good work - we really like klaxon!

**Comment by cbeust** [comment]
Thanks for the concrete example, @m4x-h4xx. I can definitely improve that!

Will keep you posted.

**Comment by cbeust** [comment]
Here is what I have so far:

```
com.beust.klaxon.KlaxonException: Unable to instantiate Person:
    Parameter numberOfEyes: expected kotlin.Int but received class java.lang.String (2)
```

**Comment by cbeust** [comment]
Fixed in 5.2.
